ci change to avoid interactive prompt

Joeavaikath opened the pull request:
## Why the changes were made

<!-- Explain why this PR is important, what problems it fixes, link related issues -->

Moves the install to within the cli e2e test suite.

## How to test the changes made

cli e2e should run without any issues. ci-Dockerfile only has kubectl install now.

<!-- Explain how the changes introduced by this PR can be tested and verified, with commands and pictures -->

kaovilai reviewed (dismissed):
Relevant changes from https://github.com/migtools/oadp-cli/pull/43

weshayutin reviewed (dismissed):
```
make install ASSUME_DEFAULT=true
Building kubectl-oadp for current platform (linux/amd64)...
✅ Built kubectl-oadp successfully!
Installing kubectl-oadp to /home/whayutin/.local/bin...
✅ Installed to /home/whayutin/.local/bin

✅ PATH already configured


📋 Configuration:
Setting Velero namespace to: openshift-adp
✅ Client config initialized

📋 Next steps:
  1. Test admin commands: kubectl oadp backup get
  2. Test non-admin commands: kubectl oadp nonadmin backup get
  3. Manage NABSL requests: kubectl oadp nabsl get
  4. Change namespace: kubectl oadp client config set namespace=<namespace>
```

Joeavaikath commented:
/retest

kaovilai commented:
/lgtm

Joeavaikath committed:
Test

Refactor OADP CLI installation in CI Dockerfile and e2e tests

- Removed direct installation of oadp-cli from the Dockerfile.
- Introduced a new CLISetup struct to handle the cloning, building, and installation of oadp-cli in a more structured manner.
- Updated e2e tests to utilize the new CLISetup for verifying OADP CLI availability and configuration.

kaovilai commented:
/retest

kaovilai commented:
```
      failed to create backup via CLI: exit status 1, output: An error occurred: namespaces "velero" not found
```

Joeavaikath commented:
> ```
>       failed to create backup via CLI: exit status 1, output: An error occurred: namespaces "velero" not found
> ```

this is weird...locally it does set the right namespace

kaovilai commented:
see more in slack DM. but maybe need this in CI?
```
❯ cat ~/.config/velero/config.json 
{"namespace":"openshift-adp"}
```

kaovilai commented:
Step e2e-test-cli-aws-e2e succeeded after 54m32s.

False alarm from flaky steps post cli e2e

/retest

mpryc commented:
/retest-required

Joeavaikath committed:
Debug

Joeavaikath committed:
Manual build and push to path

Joeavaikath committed:
Cleanup

Joeavaikath committed:
Refactor CLI binary installation process in e2e tests

- Changed the installation method from moving to copying the kubectl-oadp binary to /usr/local/bin.
- Updated log messages to reflect the change in operation.
- Improved error messages for better clarity on failure points.

Joeavaikath committed:
Update

Joeavaikath commented:
/retest

kaovilai commented on tests/e2e/lib/cli_common.go:
```diff
@@ -56,3 +59,171 @@ func ParsePhaseFromYAML(yamlOutput string) string {
+	}
+
+	// If we installed to ~/bin, ensure it's in PATH
+	if targetPath == fmt.Sprintf("%s/bin/kubectl-oadp", os.Getenv("HOME")) {
+		homeBin := fmt.Sprintf("%s/bin", os.Getenv("HOME"))
+		currentPath := os.Getenv("PATH")
+		if !strings.Contains(currentPath, homeBin) {
+			newPath := fmt.Sprintf("%s:%s", homeBin, currentPath)
+			os.Setenv("PATH", newPath)
+			log.Printf("Added %s to PATH", homeBin)
+		}
+	}
```
For the curious this is what fixed CI

weshayutin replied:
well done @kaovilai and @Joeavaikath :)

kaovilai commented:
/lgtm

Joeavaikath commented:
/retest

kaovilai commented:
I think we can stop retesting and override the hcp tests since its nil pointer deref unrelated.

kaovilai commented:
/override ci/prow/4.19-e2e-test-hcp-aws ci/prow/4.20-e2e-test-hcp-aws
